clients/src/main/java/org/apache/kafka/common/security/ssl/SslFactory.java

```diff
@@ -191,8 +191,6 @@ private void createKeystore(String type, String path, Password password, Passwor
     private void createTruststore(String type, String path, Password password) {
         if (path == null && password != null) {
             throw new KafkaException("SSL trust store is not specified, but trust store password is specified.");
-        } else if (path != null && password == null) {
-            throw new KafkaException("SSL trust store is specified, but trust store password is not specified.");
         } else if (path != null && password != null) {
```

Comment:
@granthenke The change removes the check for null password, but the truststore specified in `path` is not loaded because of the check for non-null password. I am assuming this PR is to load a specific truststore from `path` and not the default truststore (which is already supported without password). For loading truststore without password, you would also need to update `SecurityStore#load()` since it uses `password.value()`. You probably want to add an end-to-end test in `SslTransportLayerTest` since the test below didn't actually catch the fact that the configured truststore was not loaded. Also needs documentation update.